src/App.jsx

```diff
@@ -227,54 +220,64 @@ function App() {
     }
 
   }
-  function handleEdit(id) {
+  function handleAddBtn() {
     setTodoClicked(true)
+    setTodo({
+      id: id,
+      title: "",
+      timeStamp: dt,
+      description: "",
+      dueDate: "",
+      tags: "",
+      status: ""
+    })
+  }
+  function handleEdit(id) {
     setIsEditClicked(true)
+    setTodoClicked(true)
     let editTodo = data.filter(todo => todo.id === id)
     setTodo(...editTodo)
   }
 
   function handleDone() {
-    if (isEditClicked) {
-      setData(data => {
-        let newArr = [...data]
-        for (let i = 0; i < newArr.length; i++) {
-          if (newArr[i].id === todo.id) {
-            newArr[i] = { ...todo }
-          }
-        }
-        return newArr
-      })
-    }
-    else {
-      todo.tags = todo.tags ? todo.tags.split(",") : []
+    if (todo.title && todo.description && todo.status) {
+      todo.tags = todo.tags.length ? todo.tags.split(",") : []
       setTodo(todo => {
         return {
           ...todo,
           tags: todo.tags
         }
       })
+      if (isEditClicked) {
+        setData(data => {
+          let newArr = [...data]
+          for (let i = 0; i < newArr.length; i++) {
+            if (newArr[i].id === todo.id) {
+              newArr[i] = { ...todo }
+            }
+          }
+          return newArr
+        })
+      }
```

Comment:
I would also move this to a separate function. `onFinishCreate`, `onFinishEdit`, ...

Reply:
For that I will have to make a different button for edit. Right now the `Done` btn is handling the cases for both edit and add todo.